File: Codes/3. Longest Substring Without Repeating Characters.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function(s) {
   
    let maxLength=0;
    let start_=0;
    let end_=0;
    
    let charMap=new Map();
     
   while(end_<s.length){
       if(!charMap.has(s[end_])){
           charMap.set(s[end_],0);
           end_++;         
       }
       else{
            charMap.delete(s[start_]);
           start_++;
       }
       
      if(charMap.size>maxLength){
              maxLength=charMap.size;
      }  
   }
    return maxLength;
};

File: Codes/3. Longest Substring Without Repeating Characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        
        int max_length=0;
        
        HashMap<Character,Integer>m=new HashMap<>();
        
        int start=0;
        int end=0;
        
        while(end<s.length()){
            if(!m.containsKey(s.charAt(end))){
                 m.put(s.charAt(end),1);
                end++;
            }
            else{
                m.remove(s.charAt(start));
                start++;
              
            }
          if(m.size()>max_length)max_length=m.size();

        }
        
        return max_length;
        
    }
}